### DISCUSSION TITLE
Is there a way to build a preconfigured binary?

### DISCUSSION THREAD

**Discussion by jdevera** [criteo/command-launcher#135] [Q&A] [answered]
I'm trying to glue all the tooling for my homelab with the command launcher, and wonder if there is an existing way to:

1. Have a binary that comes preconfigured for autoupdate and for a command repository from the start
2. Or any other way to easily download to a new dev box and hit the ground running. (bundling commands, etc.)

**Comment by bhou** [comment, marked as answer]
For now, it is not possible to build a preconfigured binary, but the remote configuration feature can help you achieve your scenario. 

In short, it is a configuration file on a remote location, which can be accessed by all the machines that you would like to deploy command launcher. 

When command launcher starts, it will read environment variable `[PREFIX]_REMOTE_CONFIG_URL`, which points to this remote configuration file. Command launcher will merge the remote and local configurations and override the local file. 

The remote configuration file doesn't need to cover the complete configurations, you can include only the config items that you would like all command launcher instance inherits. 

For example, if you want to force them to download the same packages, you can set your remote config like this:

```json
{
   "command_repository_base_url": "https://your-remote-package-registry-root/"
}
```

Another useful configuration item related to this is the `remote_config_check_cycle`, it defined how often the command launcher checks the remote config (in hour as unit), default is 24 hours.

You can provide a bash script to setup the environment variable and download command launcher for your use case.

**Reply by jdevera** [reply]
Great, this will work better than my current approach, which is a script that makes config calls with the values.
